### PR TITLE
[6.13.z RFE] Added sca override test for Hosts

### PR DESCRIPTION
**Pull request by ColeHiggins2**
SAT-16807

Switching on SCA should not give hosts access to new custom repositories

### CODE REVIEW

**Comment by ColeHiggins2** [comment]
Wait to be mereged
 https://github.com/SatelliteQE/robottelo/pull/11380

**Comment by ColeHiggins2** [comment]
Wait to be merged 
https://github.com/SatelliteQE/airgun/pull/865

File: tests/foreman/ui/test_repositories.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+"""Unit tests for the new ``repositories`` paths.
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_repositories.py:1]:
```suggestion
""" Test module for Repositories UI.
```

File: tests/foreman/ui/test_repositories.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+
+
+@pytest.mark.rhel_ver_list([7, 8, 9])
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_repositories.py:22]:
```suggestion
@pytest.mark.rhel_ver_match('[^6]')
```

File: tests/foreman/ui/test_repositories.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    :BZ: 2188380
+    """
+    ak, org, custom_repo = setup_content
+    org.sca_disable()
+    ak.add_subscriptions(data={'subscription_id': custom_repo.product.id})
+    client = rhel_contenthost
+    client.install_katello_ca(target_sat)
+    client.register_contenthost(org.label, ak.name)
+    assert client.subscribed
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_repositories.py:55]: can you please use global registration method here to register?

File: tests/foreman/ui/test_repositories.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+
+
+@pytest.mark.rhel_ver_list('[^6]')
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_repositories.py:22]:
```suggestion
@pytest.mark.rhel_ver_match('[^6]')
```

**Review by shubhamsg199 — COMMENTED** [review]
please add `:BZ:` to the test metadata

**Comment by ColeHiggins2** [comment]
@Gauravtalreja1  @shubhamsg199 In order to see an override here, I had to not attach subscriptions and then enable sca. I then am checking that the repo was indeed "Overridden" using ```'subscription-manager repo-override --list'```

File: tests/foreman/ui/test_repositories.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    ak, org, custom_repo = setup_content
+    org.sca_disable()
+    rhel_contenthost.register(org, default_location, ak.name, target_sat)
+    assert rhel_contenthost.subscribed
+    with session:
+        session.organization.select(org.name)
+        session.location.select(default_location.name)
+        org.sca_enable()
+        repos2 = session.host_new.get_repo_sets(rhel_contenthost.hostname, custom_repo.name)
```
Comment (Griffin-Sullivan) [inline tests/foreman/ui/test_repositories.py:57]: I would prefer a more descriptive name than `repos2`. Maybe `host_repo_set` or something similar would be better.

**Review by Griffin-Sullivan — APPROVED** [review]
Ack pending comment

**Comment by jyejare** [comment]
@ColeHiggins2 @omkarkhatavkar Shouldnt we wait for the table to be appeared on the page instead of hard sleep ?

**Comment by omkarkhatavkar** [comment]
> @ColeHiggins2 @omkarkhatavkar Shouldnt we wait for the table to be appeared on the page instead of hard sleep ?

@jyejare I always in favor of wait_for, pooling sleep. I think in some cases, e.g. small sleep of 2-3 secs,  I dont want to be hard. Yes, I will be most happy to see the conditional sleep.

**Comment by jyejare** [comment]
> > @ColeHiggins2 @omkarkhatavkar Shouldnt we wait for the table to be appeared on the page instead of hard sleep ?
> 
> @jyejare I always in favor of wait_for, pooling sleep. I think in some cases, e.g. small sleep of 2-3 secs, I dont want to be hard. Yes, I will be most happy to see the conditional sleep.

Its not about small / major sleep time, its all about timing that even we could miss with these small sleep times. In fact, small sleep times are more dangerous than the bigger ones, since we get a short time to finish the task.

@ColeHiggins2 Could we go for conditional sleep please ?

**Comment by ColeHiggins2** [comment]
@jyejare @omkarkhatavkar sleep was already merged in airgun, will create seperate PR later to fix.

**Review by omkarkhatavkar — APPROVED** [review]

**Pull request merged**